src/parser/transform/helpers/transform_typename.cpp

```diff
@@ -67,6 +67,37 @@ LogicalType Transformer::TransformTypeName(duckdb_libpgquery::PGTypeName *type_n
 		D_ASSERT(children.size() == 2);
 
 		result_type = LogicalType::MAP(move(children));
+	} else if (base_type == LogicalTypeId::UNION) {
+		if (!type_name->typmods || type_name->typmods->length == 0) {
+			throw ParserException("Union needs at least one member");
+		}
+		child_list_t<LogicalType> children;
+		unordered_set<string> name_collision_set;
```

Comment:
Maybe use `case_insensitive_set_t` here? Can we have a union with `A` and `a`?

Reply:
Should I fix this for struct/map too? The same issue exist there. You can create a struct with names differing only by case, but can't insert.

```
CREATE TABLE tbl2(s STRUCT(k INT, K INT));
INSERT INTO tbl2 VALUES({'k': 10, 'k': 20});
```

Reply:
Ok, fixed for both structs and unions